src/sentry/conf/server.py

```diff
@@ -2143,6 +2143,12 @@ def create_partitioned_queues(name):
 
 SENTRY_REPROCESSING_SYNC_REDIS_CLUSTER = "default"
 
+# Timeout for the project counter statement execution.
+# In case of contention on the project counter, prevent workers saturation with
+# save_event tasks from single project.
+# Set to `None` to disable.
+SENTRY_PROJECT_COUNTER_STATEMENT_TIMEOUT = 3000
```

Comment:
Wondering if 3 seconds is actually too long here. I feel like if we're blocking workers, a bunch of things blocking for 3 seconds might still be bad.

Reply:
Lowering this down to 1s. We have metrics for this that show that in normal operations call to project counter increment takes on avg 2ms.